src/networking.c

```diff
@@ -3016,12 +3020,18 @@ void readQueryFromClient(connection *conn) {
     }
 
     if (c && (c->io_flags & CLIENT_IO_REUSABLE_QUERYBUFFER)) {
-        serverAssert(c->qb_pos == 0); /* Ensure the client's query buffer is trimmed in processInputBuffer */
```

Comment:
@sundb for io thread, as said in top comment, we may defer the trim operation of query buffer, so this assert may fail, this also means the shared query buffer can not be reused. please have a look.